pkg/kubelet/network/dns/dns.go

```diff
@@ -273,7 +273,7 @@ func parseResolvConf(reader io.Reader) (nameservers []string, searches []string,
 			}
 		}
 		if fields[0] == "options" {
-			options = fields[1:]
+			options = appendOptions(options, fields[1:]...)
```

Comment:
why do we need to parse original option list for every line?

Reply:
https://github.com/kubernetes/kubernetes/issues/112323#issuecomment-1253128397
> I found that there is a logic to merging DNS options in the pod DNS spec generating process. So I followed the logic.
> 
> If the pod is using `dnsPolicy:Default` with `dnsConfig.options`, the current kubelet will merge the node's last option line in `/etc/resolv.conf` with `dnsConfig.options`.


https://github.com/kubernetes/kubernetes/blob/127f33f63d118d8d61bebaba2a240c60f71c824a/pkg/kubelet/network/dns/dns.go#L325-L353

Reply:
Use `strings.Cut` instead.